starboard/tools/abstract_launcher.py

```diff
@@ -95,8 +96,42 @@ def LauncherFactory(platform_name,
       **kwargs)
 
 
+class ReturnCodeStatus(IntEnum):
+  """Represents status of code, returned by target"""
+
+  #: Target exited normally. Return code is available
+  OK = 0
+
+  #: Target exited normally. Return code is not available
+  NA = 1
+
+  #: Target crashed
+  CRASH = 2
+
+  #: Target not started
+  ISSUE = 3
+
+
 class AbstractLauncher(object):
-  """Class that specifies all required behavior for Cobalt app launchers."""
+  """
+  Class that specifies all required behavior for Cobalt app launchers.
+  The following is definition of extended interface. Implementation is optional
+  Functions:
+    - InitDevice: a function to be called before any other performance
+      on the target device
+    - RebootDevice: a function, used to reboot the target device
+    - Run2: a function to run a target on device. Must be implemented
```

Comment:
Should we add this as an abstract method as well below ?

Reply:
My idea is that a user of extended interface can guess about underlying implementation by testing it with `hasattr` function. In case of abstract methods `hasattr` has no effect and it's a bit harder for the user to discover details of underlying implementation.